### ISSUE TITLE
Feature: Use keyless authentication

### ISSUE THREAD

**Issue by JPaivBerops** [Berops/claudie#149]
### Description:

I saw that we using service account email + service account key to authenticate with Google Cloud. I consider this to be a security issues for following reasons:

1. Service key is permanent, in the wrongs hands it can do a lot of damage.

2. It is constant until we disable it so we need to always be on guard.

3. We have to worry about it's permanent storage and storage encryption. 

4. We have to manage the service key rotation. 

Also when I ran the GHA pipeline the console input told me that the authentication method is deprecated. The deprecation message proposed using keyless authentication.

**So what is keyless authentication?**

In short keyless authentication is about generating a temporal short-lived OAuth 2.0 or JWT credential for deployment. Usually these last an hour and are only used by the deployment pipeline. More are generated when needed so we don't reuse existing tokens. 

**How keyless auth works?**

Quite simply like this: https://cloud.google.com/blog/products/identity-security/enabling-keyless-authentication-from-github-actions

There is a premade tool by google so setting it up is easy.

### Exit criteria:
- [ ] Provision workload identity Federation to Google Cloud
- [ ] Replace currently login with Workload identity based auth on GHA pipelines
